### PR TITLE
Replace area-wide fire overlay with reddening the lights

### LINKED ISSUE #35415: Remove the bad red overlay from fire alarms and replace them with emergency lights
firefox_2018-02-07_15-19-27
tldr remove the flashing red overlay from fire alarms and replace them with light tubes and bulbs turning into a red hue when the fire alarm goes off

@vuonojenmustaturska gave permission to make this ~~feature request~~ improvement request

### PR DESCRIPTION
**Pull request by SpaceManiac**
:cl:
tweak: Fire alarms now redden all the room's lights, and emit light themselves, rather than applying an area-wide overlay.
fix: Fire alarms no longer visually conflict with radiation storms.
/:cl:

Fixes #35415.

|**Old Appearance**|**New Appearance**|
|-|-|
| space station 13 2018-06-08 141618 | ss13 superbox station 13 2018-06-08 141003 |
| space station 13 2018-06-08 141645 | ss13 superbox station 13 2018-06-08 141054 |
| space station 13 2018-06-08 141654 | ss13 superbox station 13 2018-06-08 141103 |
| space station 13 2018-06-08 142952 | ss13 superbox station 13 2018-06-08 143157 |

## Originals for reference
ss13 superbox station 13 2018-06-08 141239
ss13 superbox station 13 2018-06-08 141214

### CODE REVIEW

**Comment by PKPenguin321** [comment]
But won't fire just blow out all the lights in 5 seconds

**Comment by cyclowns** [comment]
then you have another light source for letting you know theres a fire

a fire

File: code/game/machinery/firealarm.dm [hunk 1]
```diff
@@ -273,35 +278,10 @@
...
-		return
-	A.partyalert()
-
-/obj/machinery/firealarm/partyalarm/ui_data(mob/user)
-	. = ..()
-	var/area/A = get_area(src)
-	.["alarm"] = A.party
+/obj/machinery/firealarm/proc/update_fire_light(fire)
+	if(fire == !!light_power)
+		return  // do nothing if we're already active
```
Comment (Cyberboss) [inline code/game/machinery/firealarm.dm]: This is confusing to read, should probably just be dropped since set_light() already has code for this iirc

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/machinery/firealarm.dm (hunk 1 above)
Comment (SpaceManiac) [reply code/game/machinery/firealarm.dm]: It still gets queued, only the heavy part of the update is skipped when SSlighting eventually gets to it.

**Pull request merged**